File: Python/Container With Most Water .py
class Solution:
    # @return an integer
    def maxArea(self, height):
        j = len(height) - 1
        i = 0        
        maxium = 0

        while i < j:
            maxium = max( maxium, (j-i) * min(height[i], height[j]))
            if height[i] < height[j]:
                i += 1
            elif height[i] >= height[j]:
                j -=1
        return maxium

File: Java/src/Container With Most Water .java
public class Solution
{
  public int maxArea(int[] height)
  {
    int i = 0;
    int j = height.length - 1;
    int maxium = 0;
    while (i < j)
    {
      maxium = Math.max(maxium, (j - i) * Math.min(height[i], height[j]));
      if (height[i] < height[j])
      {
        i++;
      }
      else
      {
        j--;
      }
    }
    return maxium;
  }
}
